control-plane/agents/src/bin/core/controller/scheduling/mod.rs

```diff
@@ -168,6 +168,18 @@ impl ChildSorters {
         match Self::sort_by_health(a, b) {
             Ordering::Equal => match Self::sort_by_child(a, b) {
                 Ordering::Equal => {
+                    // Remove replicas from nodes which are cordoned with most priority.
+                    if let (Some(a), Some(b)) = (a.node_spec(), b.node_spec()) {
```

Comment:
Looks this this spec is optional, meaning a replica may point to a node which does not exist anymore?
In such case should a replica without node spec always be preferred to be removed?

Reply:
Okay, we can do that.